azureml/__init__.py

```diff
@@ -803,7 +803,7 @@ def __getitem__(self, index):
                 if experiment['ExperimentId'] == index:
                     return self._create_experiment(experiment)
 
-        raise IndexError()
+        raise IndexError('An experiment named "{}" does not exist'.format(index))
```

Comment:
An experiment with id "{}"

There's no such thing as an experiment name, just 'description' and 'id'.